### ISSUE TITLE
Support "forward references" of steps in factfiles

### ISSUE THREAD

**Issue by ninjabear** [snowplow/factotum#31]
Currently a step can only depend on a step which the parser has previously seen. For example;

    step a
    step b 
    step c <-- can only depend on steps b or a, not d
    step d   

This simplifies the parser logic, but restricts the file layout somewhat. Factotum should also not panic if this kind of event occurs (which as of 0.1.0 it does).

**Comment by alexanderdean** [comment]
Good point, nice to do. We should track limitations in the README for now (Chronos style :goberserk: )
